lib/core/src/Cardano/Wallet/Primitive/AddressDerivation.hs

```diff
@@ -202,6 +209,32 @@ instance ToText Role where
 instance FromText Role where
     fromText = fromTextToBoundedEnum SnakeLowerCase
 
+-- | Bring a 'Role' type back to the term-level. This requires a type
+-- application and either a scoped type variable, or an explicit passing of a
+-- 'Role'.
+--
+-- >>> roleVal @'UtxoExternal
+-- UtxoExternal
+--
+-- >>> roleVal @chain
+-- ...
+roleVal :: forall (c :: Role). Typeable c => Role
```

Comment:
👍 For moving this function here.

(I want to replace it to use a proper type class instead of `Typeable` in some future, but not now.)